File: testSuite/test-extract-directives.pl
#!/usr/bin/env perl
use strict;
use warnings;
use Cwd;
use lib $ENV{'GALACTICUS_EXEC_PATH'}."/perl";
use Galacticus::Build::Directives;
use Data::Dumper;

# Test extraction of directives from source files.
# Andrew Benson (10-June-2021)

my @directives = &Galacticus::Build::Directives::Extract_Directives($ENV{'GALACTICUS_EXEC_PATH'}."/testSuite/data/sourceWithDirectives.F90","*",setRootElementType => 1);
my $status = scalar(@directives) == 10 ? "success" : "FAILED";
if ( $status eq "success" ) {
    print $status.": all directives found\n";
} else {
    print $status.": all directives NOT found\n";
}

exit;

File: testSuite/test-extract-directives.py
#!/usr/bin/env python3
import subprocess
import sys
import os

# Test extraction of directives from source files.
# Andrew Benson (ported to Python)

execPath = os.environ.get("GALACTICUS_EXEC_PATH", "..")

# Run the Perl module via subprocess.
result = subprocess.run(
    f'perl -e \'use lib "{execPath}/perl"; use Galacticus::Build::Directives; '
    f'my @directives = &Galacticus::Build::Directives::Extract_Directives("{execPath}/testSuite/data/sourceWithDirectives.F90","*",setRootElementType => 1); '
    f'my $status = scalar(@directives) == 10 ? "success" : "FAILED"; '
    f'if ($status eq "success") {{ print $status.": all directives found\\n"; }} '
    f'else {{ print $status.": all directives NOT found\\n"; }}\'',
    shell=True
)
